controllers/confidential_handler.go

```
@@ -92,12 +97,19 @@ func (r *KataConfigOpenShiftReconciler) handleConfidentialPeerPods(state Feature
 
 	// Patch peer pods configMap, if it exists.
 	var peerpodsCMData map[string]string
+	var keysToRemove map[string]string
 	if state == Enabled {
 		peerpodsCMData = map[string]string{"DISABLECVM": "false"}
+		// Remove INITDATA if it matches the default value
+		keysToRemove = map[string]string{"INITDATA": defaultNonCCInitdata}
```

This will silently go against the intent of the user if they want to set such a permissive INITDATA on purpose... is it something we want ?

As said earlier, this is an ownership problem. It doesn't look right to tell the user they can set INITDATA if OSC is likely to overwrite it. It looks like user input should go to a dedicated CM or at least to a separate CUSTOM_INITDATA entry in the peerpods CM.

Cc @pmores for insights.

---

> This will silently go against the intent of the user if they want to set such a permissive INITDATA on purpose... is it something we want ? 

It's very unlikely to collide as CoCo users will normally be required to set in the initdata not only the agent policy but also aa/cdh toml, having said that, i can add a unique comment in the file "This initdata was created by the osc-operator" or something like that.

>  It doesn't look right to tell the user they can set INITDATA if OSC is likely to overwrite it.

Bare in mind we set it only for the non-CoCo users while initdata is officially supported only in CoCo. (FWIW, the concept of dummy (empty) initdata for pp is used also upstream).

At this point i find this solution fairly decent, however, I agree it's an ownership problem and open to better option if someone have it.

---

This ownership problem requires some redesign that we might not be able to perform this close to the release. Let's add a `This initdata was created by the osc-operator` marker inside the OSC generated one in the meantime then.

---

In general I believe (as I said before) that every k8s resource should have a single owner.  It shouldn't be that different items of the same resource have different owners, let alone that that a single item has multiple owners, the reason being that it can get confusing and frustrating rather quickly.

In particular, I unfortunately don't have enough context to propose a specific solution here but I do believe that if we let users set something we should do our best to avoid overwriting it later.  I understand that the semantics of the current situation seemingly makes incidents unlikely.  I still feel there should be ways to do better - e.g. if I understand correctly the setting in question here only makes sense under certain circumstances.  This sounds like something to be enforced by a schema.